cocotb/share/makefiles/Makefile.inc

```diff
@@ -29,20 +29,18 @@
 
 # Common makefile included by everything
 
-# Directory containing the cocotb Python module
-ifeq ($(COCOTB_PY_DIR),)
-export COCOTB_PY_DIR:=$(realpath $(dir $(lastword $(MAKEFILE_LIST)))/../../..)
+# Make sure cocotb is installed
+COCOTB_CONFIG_CMD := $(shell :; command -v cocotb-config)
+ifeq ($(COCOTB_CONFIG_CMD),)
+    $(error "Cocotb not installed (cocotb-config not found). Please install cocotb (see documentation for details).")
 endif
```

Comment:
I think this belongs in the other `Makefile.inc`. If the only way users are including this file is via `cocotb-config --makefiles`, then the fact this was included at all is an indication that it's already installed.

Reply:
In this case, this https://github.com/cocotb/cocotb/pull/1445/files#diff-9b99a3d072edd3bd56cffb557c7d942fR33 would need to be an error and sing other Makefile.inc would be not possible (ok for me).

Reply:
It's already not possible to use the other makefile. One of three cases happens as you have this patch:

1. Cocotb is not installed. Whether the user uses old or new `Makefile.inc`, they're going to hit the error
2. Cocotb is a regular install. The old `Makefile.inc` includes the new `Makefile.inc` from the source tree, which then goes on to use submakefiles from the installation
3. Cocotb is a development install. The above happens, but installation and source tree are equivalent.

Case 2 is just confusion waiting to happen. If old `Makefile.inc` always redirects to the _installed_ new `Makefile.inc`, case 2 becomes much more sensible.